src/components/common/CreateUserWizard/StepUserName.tsx

```diff
@@ -54,13 +55,25 @@ const UsernameInput = ({ disabled }: UsernameInputProps) => (
 type Props = WizardStepProps<FormValues, UserWizardStep2>;
 
 const StepUserName = ({
+  nextStep,
   wizardValues,
   wizardForm: { initialValues: defaultValues },
 }: Props) => {
   const navigate = useNavigate();
+  const { pathname } = useLocation();
+  const { wallet, updateUser } = useAppContext();
   const transform = mergePayload(wizardValues);
+  /* Handle if user flow is within Colony create */
   /* Replace: true so you can't get back to the User wizard after completion */
-  const handleSuccess = () => navigate(LANDING_PAGE_ROUTE, { replace: true });
+  const handleSuccess = () => {
+    if (!matchPath(pathname, CREATE_USER_ROUTE)) {
+      if (updateUser && wallet) {
+        updateUser(wallet?.address);
+      }
+      return nextStep(wizardValues);
+    }
```

Comment:
Re my previous comment, here you could check for the presence of the flag and redirect them to colony creation